server/sonar-server/src/main/java/org/sonar/server/platform/Platform.java

```diff
@@ -78,14 +82,29 @@ public void init(Properties properties, ServletContext servletContext) {
 
   // Platform is injected in Pico, so do not rename this method "start"
   public void doStart() {
-    if (!started && getDatabaseStatus() == DatabaseVersion.Status.UP_TO_DATE) {
+    if (started && !isInSafeMode()) {
+      return;
+    }
+
+    if (getDatabaseStatus() == DatabaseVersion.Status.UP_TO_DATE) {
       startLevel34Containers();
-      started = true;
+      if (isInSafeMode()) {
```

Comment:
Not sure that "safe mode" is clear enough. Don't you think that name should mention the fact that db is connected but schema is not up-to-date ?